tests/system/Database/BaseQueryTest.php

```diff
@@ -420,6 +420,163 @@ public function testSetQueryBinds()
         $this->assertSame($expected, $query->getQuery());
     }
 
+    public function testGetQueryMultipleTimes()
+    {
+        $query = new Query($this->db);
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name:', ['name' => 'placeholder :name: in value']);
+
+        $expected = "SELECT * FROM users WHERE name = 'placeholder :name: in value'";
+
+        // Triggers compileBinds()
+        $query->getQuery();
+
+        $this->assertSame($expected, $query->getQuery());
+    }
+
+    public function testSetBindsMultipleTimes()
+    {
+        $query = new Query($this->db);
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name:', ['name' => 'John']);
+
+        // Triggers compileBinds()
+        $query->getQuery();
+
+        $query->setBinds(['name' => 'Jane']);
+
+        $expected = "SELECT * FROM users WHERE name = 'Jane'";
+
+        $this->assertSame($expected, $query->getQuery());
+    }
+
+    public function testSetQueryMultipleTimesKeepingBinds()
+    {
+        $query = new Query($this->db);
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name:', ['name' => 'John']);
+
+        // Triggers compileBinds()
+        $query->getQuery();
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name: OR id = 1');
+
+        $expected = "SELECT * FROM users WHERE name = 'John' OR id = 1";
+
+        $this->assertSame($expected, $query->getQuery());
+    }
+
+    public function testSetQueryMultipleTimesReplacingBinds()
+    {
+        $query = new Query($this->db);
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name:', ['name' => 'John']);
+
+        // Triggers compileBinds()
+        $query->getQuery();
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name: OR id = 1', ['name' => 'Jane']);
+
+        $expected = "SELECT * FROM users WHERE name = 'Jane' OR id = 1";
+
+        $this->assertSame($expected, $query->getQuery());
+    }
+
+    public function testSetQueryMultipleTimesRemovingBinds()
+    {
+        $query = new Query($this->db);
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name:', ['name' => 'John']);
+
+        // Triggers compileBinds()
+        $query->getQuery();
+
+        $query->setQuery('SELECT * FROM users WHERE name = :name: OR id = 1', []);
+
+        $expected = 'SELECT * FROM users WHERE name = :name: OR id = 1';
+
+        $this->assertSame($expected, $query->getQuery());
+    }
+
+    public function testSwapPrefixMultipleTimes()
+    {
+        $query = new Query($this->db);
+
+        $origPrefix = 'db_';
+        $newPrefix1 = 'ci_';
+        $newPrefix2 = 'zz_';
+
+        $origSQL = 'SELECT * FROM db_users WHERE db_users.id = 1';
+        $newSQL  = 'SELECT * FROM zz_users WHERE zz_users.id = 1';
+
+        $query->setQuery($origSQL);
+        $query->swapPrefix($origPrefix, $newPrefix1);
+        $query->swapPrefix($newPrefix1, $newPrefix2);
```

Comment:
I think there is no use case that calls swapPrefix() more than once.

Reply:
It is possible to swap prefix several times with the current code, so I made sure it was still possible with my new code, just in order to not introduce any regression.

Still, it should be worth covering this with a test, to ensure the behaviour is consistent.

Reply:
Okay.